ros2doctor/ros2doctor/api/network.py

```diff
@@ -87,12 +176,44 @@ def category(self):
 
     def report(self):
         """Print system and ROS network information."""
-        # check ifcfg import for windows and osx users
-        ifcfg_ifaces = ifcfg.interfaces()
+        if_stats = psutil.net_if_stats()
 
         network_report = Report('NETWORK CONFIGURATION')
-        for iface in ifcfg_ifaces.values():
-            for k, v in iface.items():
+
+        for interface, addrs in psutil.net_if_addrs().items():
+            if_info = {
+                'inet': None,
+                'inet4': [],
+                'ether': None,
+                'inet6': [],
+                'netmask': None,
+                'device': interface,
+                'flags': None,
+                'mtu': None,
+                'broadcast': None,
```

Comment:
This dict seems like a prime use case for [dataclasses](https://docs.python.org/3/library/dataclasses.html) would you accept a patch that creates and uses an InterfaceInfo dataclass instead of a dict or would that be gilding the lily?

Reply:
On further reading since these values are added to the report in raw key-value pairs we'd have to tease the dataclass apart anyway https://github.com/ros2/ros2cli/pull/687/files/e4b540a3f1aa10215a7f7064b26281b7331a914d#diff-9fcdc075292a69769ee336433dd05af791b6a8363dc138ac06fe852ac5aed05aR216-R218

Reply:
> On further reading since these values are added to the report in raw key-value pairs we'd have to tease the dataclass apart anyway

Right.  This way seems pretty straightforward to me (though not the shortest of code), so I'm inclined to leave it unless there is another way that is clearer.